noir/noir-repo/noir_stdlib/src/hash/poseidon2.nr

```diff
@@ -16,7 +16,7 @@ impl Poseidon2 {
         Poseidon2::hash_internal(input, message_size, message_size != N)
     }
 
-    pub(crate) fn new(iv: Field) -> Poseidon2 {
+    pub fn new(iv: Field) -> Poseidon2 {
```

Comment:
@TomAFrench you fine with this?

I feel like given that Poseidon2 struct is pub it's fine to have this be pub as well.